troubadix/helper/linguistic_exception_handler.py

```diff
@@ -0,0 +1,155 @@
+        self.pattern = re.compile(pattern, flags=flags)
+
+    def execute(self, file: str, content: str):
+        return bool(self.pattern.search(content))
+
+
+class PatternsCheck(LinguisticCheck):
+    def __init__(
+        self,
+        patterns: Union[List[str], List[Tuple[str, re.RegexFlag]]],
+    ) -> None:
+        # Originally I tried isinstance(patterns, Tuple[...]) for testing,
+        # but that throws an exception because Tuple[...]
+        # is only valid for type hinting :/
+        if isinstance(patterns[0], Tuple):
+            self.patterns = [
+                re.compile(pattern, flags=flags) for pattern, flags in patterns
+            ]
+        else:
+            self.patterns = [re.compile(pattern) for pattern in patterns]
+
+    def execute(self, file: str, content: str):
+        return any(bool(pattern.search(content)) for pattern in self.patterns)
+
+
+class CompositeCheck(LinguisticCheck):
+    def __init__(self, *checks: Iterable[LinguisticCheck]) -> None:
+        self.checks = checks
+
+    def execute(self, file: str, content: str):
+        return all(check.execute(file, content) for check in self.checks)
+
+
+class TextInFileCheck(CompositeCheck):
+    def __init__(self, file: str, text: str) -> None:
+        super().__init__(FileCheck(file), TextCheck(text))
+
+
+class PatternInFileCheck(CompositeCheck):
+    def __init__(
+        self, file: str, pattern: str, flags: re.RegexFlag = 0
+    ) -> None:
+        super().__init__(FileCheck(file), PatternCheck(pattern, flags))
+
+
+class PatternsInFileCheck(CompositeCheck):
+    def __init__(
+        self,
+        file: str,
+        patterns: Union[List[str], List[Tuple[str, re.RegexFlag]]],
+    ) -> None:
+        super().__init__(FileCheck(file), PatternsCheck(patterns))
+
+
+class PatternInFilesCheck(CompositeCheck):
+    def __init__(
+        self, files: List[str], pattern: str, flags: re.RegexFlag = 0
+    ) -> None:
+        super().__init__(FilesCheck(files), PatternCheck(pattern, flags))
+
+
+class PatternInFilePatternCheck(CompositeCheck):
+    def __init__(
+        self,
+        file_pattern: str,
+        text_pattern: str,
+        file_pattern_flags: re.RegexFlag = 0,
+        text_pattern_flags: re.RegexFlag = 0,
+    ) -> None:
+        super().__init__(
+            FilePatternCheck(file_pattern, file_pattern_flags),
+            PatternCheck(text_pattern, text_pattern_flags),
+        )
+
+
+class PatternsInFilePatternCheck(CompositeCheck):
+    def __init__(
+        self,
+        file_pattern: str,
+        patterns: Union[List[str], List[Tuple[str, re.RegexFlag]]],
+        file_pattern_flags: re.RegexFlag = 0,
+    ) -> None:
+        super().__init__(
+            FilePatternCheck(file_pattern, file_pattern_flags),
+            PatternsCheck(patterns),
+        )
+
+
+def handle_linguistic_checks(
+    file: str, content: str, checks: Iterable[LinguisticCheck]
```

Comment:
```suggestion
    file: str, correction: str, checks: Iterable[LinguisticCheck]
```
Just like `_parse_codespell_line`, can we rename this is to `correction`. Same goes for all other occurrences in this file.
This is more expressive compared to generic `content` IMHO, because when reading `file` and `content`, the first thing I though of was file content.

Reply:
Renamed